tests/lib/types/dag-merge.nix

```diff
@@ -22,6 +22,9 @@ in {
         between =
           mkIf true (dag.entryBetween [ "after" ] [ "before" ] "between");
       }
+      { last = "three"; }
+      { last = dag.entryAfter [ "after" ] "two"; }
+      { last = dag.entryAfter [ "before" ] "one"; }
```

Comment:
This test doesn't convince me that this actually works as intended. The test should ensure that the values are actually merged. In this case, it could be that they weren't actually merged correctly since it's at the end of the list anyway.

Maybe if you bounded it on both sides it would be more convincing? That is, having a merged `dag.entryBefore` on `last` as well that bounds where this can be in the output.

Also, how does this work with `entryBetween`?

Reply:
I've added both an `entryBetween` and an `entryBefore` to test this better. (Note that `entryBetween` has its arguments flipped; this really means "entry between before and after")